zebrad/src/config.rs

```diff
@@ -54,3 +71,80 @@ pub struct ZebradConfig {
+    /// 1. Hard-coded defaults (lowest precedence)
+    /// 2. TOML configuration file (if provided)
+    /// 3. Environment variables with `ZEBRA_` prefix (highest precedence)
+    ///
+    /// Environment variables use the format `ZEBRA_SECTION__KEY` where:
+    /// - `SECTION` is the configuration section (e.g., `network`, `rpc`)
+    /// - `KEY` is the configuration key within that section
+    /// - Double underscores (`__`) separate nested keys
+    ///
+    /// # Security
+    /// Environment variables whose leaf key names end with sensitive suffixes (case-insensitive)
+    /// will cause configuration loading to fail with an error: `password`, `secret`, `token`, `cookie`, `private_key`.
+    /// This prevents both silent misconfigurations and process table exposure of sensitive values.
+    ///
+    /// # Examples
+    /// - `ZEBRA_NETWORK__NETWORK=Testnet` sets `network.network = "Testnet"`
+    /// - `ZEBRA_RPC__LISTEN_ADDR=127.0.0.1:8232` sets `rpc.listen_addr = "127.0.0.1:8232"`
+    pub fn load(config_path: Option<PathBuf>) -> Result<Self, config::ConfigError> {
+        let mut builder = config::Config::builder();
+
+        // 1. Start with defaults - but don't use try_from with the struct directly
+        // Instead, we'll let config-rs use its own defaults and override as needed
+
+        // 2. Load from TOML file if provided
+        if let Some(path) = config_path {
+            builder = builder.add_source(config::File::from(path).required(true));
+        }
+
+        // 3. Load from standard ZEBRA_ environment variables with a sensitive-leaf deny-list
+        // Use ZEBRA_ prefix and __ as separator for nested keys
+        // We filter the raw environment first, then let config-rs parse types via try_parsing(true).
+        // Sensitive environment variables cause configuration loading to fail for security.
+        let mut filtered_env: std::collections::HashMap<String, String> =
+            std::collections::HashMap::new();
+        for (key, value) in std::env::vars() {
+            if !key.starts_with("ZEBRA_") {
+                continue;
+            }
+
+            // Strip the prefix and split nested keys on "__" to find the leaf key name.
+            if let Some(without_prefix) = key.strip_prefix("ZEBRA_") {
+                let parts: Vec<&str> = without_prefix.split("__").collect();
+                if let Some(leaf) = parts.last() {
+                    if is_sensitive_leaf_key(leaf) {
+                        return Err(config::ConfigError::Message(format!(
+                            "Environment variable '{}' contains sensitive key '{}' which cannot be overridden via environment variables. \
+                             Use the configuration file instead to prevent process table exposure.",
+                            key, leaf
+                        )));
+                    }
+                }
+            }
+
+            filtered_env.insert(key, value);
+        }
```

Comment:
Optional/Subjective/Nitpick:

```suggestion
        for (key, value) in std::env::vars().filter(|(key, _)| !key.starts_with("ZEBRA_")) {
            let leaf = key.split("_").last().expect("key must start with 'ZEBRA_'");
            if is_sensitive_leaf_key(leaf) {
                let err_msg = format!(
                    "Environment variable '{key}' contains sensitive key '{leaf}' which \
                     cannot be overridden via environment variables. \
                     Use the configuration file instead to prevent process table exposure.",
                );
                return Err(config::ConfigError::Message(err_msg));
            }

            filtered_env.insert(key, value);
        }
```

Reply:
It seems like this would split on a single underscore (`_`). For a variable like `ZEBRA_RPC__LISTEN_ADDR`, `key.split("_")` would produce `["ZEBRA", "RPC", "", "LISTEN", "ADDR"]`, making the leaf key `"ADDR"` instead of `"LISTEN_ADDR"`